msu/systems/mod_settings/mod_settings_system.nut

```diff
@@ -79,17 +79,24 @@
 		/*
 		_data = {
 			modID = {
-				settingID = value
+				settingID =
+				{
+					type,
+					value
+				}
 			}
 		}
 		*/
-
 		foreach (modID, panel in _data)
 		{
-			foreach (settingID, value in panel)
+			foreach (settingID, data in panel)
 			{
-				local setting = this.Panels[modID].getSetting(settingID);
-				setting.set(value);
+				this.updateSettingFromJS({
+					mod = modID,
+					id = settingID,
+					type = data.type,
+					value = data.value
+				});
```

Comment:
I don't like how this calls the JS function, when it might as well be called from SQ, Probably just add an universial updateSetting function.

Reply:
This function is called from JS though, maybe this function should be renamed to updateSettingsFromJS instead, I'd prefer that approach